src/Http/Http.Results/src/HttpResultsHelper.cs

```diff
@@ -30,7 +30,7 @@ internal static partial class HttpResultsHelper
 
         Log.WritingResultAsJson(logger, typeof(T).Name);
 
-        return httpContext.Response.WriteAsJsonAsync(
+        return httpContext.Response.WriteAsJsonAsync<object?>(
```

Comment:
Could you explain this change

Reply:
Providing the `object` as a type will force the Json Serializer check the runtime type and we will have the consistency in the serialization for both `TypedResults` and `Results`. Also, it is the same what we do when return the type directly instead of a `IResult`.

Reply:
This does have the unfortunate consequence of forcing a struct to be boxed, whereas before it would implicitly use `WriteAsJsonAsync<T>` and not box.

Reply:
Actually, I might have an idea to avoid the boxing. I will test it and update PR.

Reply:
@BrennanConroy  I added a condition to avoid the boxing when we are serializing a `ValueType` since polymorphism is not supported.